flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/ReadWriteTableITCase.java

```diff
@@ -388,220 +437,1131 @@ public void testBatchWriteNonPartitionedRecordsWithoutPk() throws Exception {
+        collectAndCheck(
+                tEnv,
+                managedTable,
+                Collections.emptyMap(),
+                "hh = '00' OR hh = '12'",
+                Arrays.asList(
+                        changelogRow("+I", "HK Dollar", 80L, "2022-01-01", "00"),
+                        changelogRow("+I", "Euro", 120L, "2022-01-02", "12")));
+
+        // streaming iter will not receive any changelog
+        assertNoMoreRecords(streamIter);
+
+        // test partition filter
+        collectAndCheckBatchReadWrite(
+                Arrays.asList("dt", "hh"), // partition
+                Arrays.asList("currency", "dt", "hh"), // pk
+                "hh >= '10'",
+                Collections.emptyList(),
+                Arrays.asList(
+                        // dt = 2022-01-01, hh = 20
+                        changelogRow("+I", "Yen", 1L, "2022-01-01", "20"),
+                        changelogRow("+I", "Euro", 114L, "2022-01-01", "20"),
+                        changelogRow("+I", "US Dollar", 114L, "2022-01-01", "20"),
+                        // dt = 2022-01-02, hh = 12
+                        changelogRow("+I", "Euro", 119L, "2022-01-02", "12")));
+
+        // test field filter
+        collectAndCheckBatchReadWrite(
+                Arrays.asList("dt", "hh"), // partition
+                Arrays.asList("currency", "dt", "hh"), // pk
+                "rate >= 119",
+                Collections.emptyList(),
+                Collections.singletonList(
+                        // dt = 2022-01-02, hh = 12
+                        changelogRow("+I", "Euro", 119L, "2022-01-02", "12")));
+
+        // test projection
+        collectAndCheckBatchReadWrite(
+                Arrays.asList("dt", "hh"), // partition
+                Arrays.asList("currency", "dt", "hh"), // pk
+                null,
+                Collections.singletonList("hh"),
+                Arrays.asList(
+                        changelogRow("+I", "00"), // Yen, 1L, 2022-01-01
+                        changelogRow("+I", "00"), // Euro, 114L, 2022-01-01
+                        changelogRow("+I", "00"), // US Dollar, 114L, 2022-01-01
+                        changelogRow("+I", "20"), // Yen, 1L, 2022-01-01
+                        changelogRow("+I", "20"), // Euro, 114L, 2022-01-01
+                        changelogRow("+I", "20"), // US Dollar, 114L, 2022-01-01
+                        changelogRow("+I", "12") // Euro, 119L, "2022-01-02
+                        ));
+
+        // test projection and filter
+        collectAndCheckBatchReadWrite(
+                Arrays.asList("dt", "hh"), // partition
+                Arrays.asList("currency", "dt", "hh"), // pk
+                "rate > 100 AND hh >= '20'",
+                Collections.singletonList("rate"),
+                Collections.singletonList(changelogRow("+I", 114L)));
+    }
+
+    @Test
+    public void testBatchWriteMultiPartitionedRecordsWithoutPk() throws Exception {
```

Comment:
Can we consider putting the new tests inside another class and shortening those method names? I think this class is too big and method name is long.